content/1_docs/2_cookbook/3_templating/0_understanding-oop/recipe.txt

```diff
@@ -0,0 +1,462 @@
+## Namespaces
+
+In larger applications, we often make use of third party libraries so as not having to reinvent the wheel.
+
+When we do so, it might happen that these third party libraries use the same class names as we do in our own code, which can easily result in errors, because classes like functions cannot be redeclared multiple times.
+
+To prevent such collisions, classes usually live in their own namespaces, so that even if a class has the same name, there will be no conflicts. Namespaces are also a good way to group different classes logically.
+
+Let's add a namespace to our `Book` and `Ebook` classes, for example `Cookbook\Books`:
+
+```php "/book/book.php"
+<?php
+
+namespace Cookbook\Books;
+
+class Book {
+    //… code as before
+}
+```
+
+```php "/book/ebook.php"
+<?php
+
+namespace Cookbook\Books;
+
+class EBook {
+    //… code as before
+}
+```
+
+If we open `index.php` after having made this change, we will be greeted with an error: "Fatal error: Uncaught Error: Class "Book" not found in …/index.php on line ….
+
+That is because the full name of the `Book` class is now `Cookbook\Books\Book` and `Cookbook\Books\Ebook` respectively.
+
+We can now either add the full classname when invoking our objects:
+
+```php "/book/index.php"
+$book = new Cookbook\Books\Book('The Hitchhiker\'s Guide to the Galaxy', 'Douglas Adams');
+```
+
+or we import the class…
+
+### Importing classes
+
+Instead of instantiating an object with its full class name, we can import the classes using the `use` keyword. Once that is done, we can continue using the short name when instatiation an object like before:
+
+
+```php "/book/index.php"
+<?php
+// make PHP display errors on screen
+ini_set('display_errors', 1);
+// require the book class
+require_once __DIR__ . '/book.php';
+require_once __DIR__ . '/ebook.php';
+
+use Cookbook\Books\Book;
+use Cookbook\Books\EBook;
+
+$book  = new Book(…);
+$ebook = new EBook(…);
+//…
+```
+
+## Static properties and methods
+
+Static properties and methods can be called directly without creating an instance of a class using the `new` keyword.
+
+You have probably come around such static methods when using Kirby. Same classes in the Toolkit, for example the `A`, Str` or 'Html` classes, have only static methods.
+
+That we don't have to explicitly instanciate an object is not the only difference, we also call these methods differently, for example:
+
+```php
+echo Str::upper('Kirby is my favorite CMS');
+```
+
+Instead of the arrow syntax used for methods that are called on an instance of a class like `$page->update()`, a double colon (or (link: https://en.wikipedia.org/wiki/Scope_resolution_operator text: "Paamayim Nekudotayim")) is used to access a static method or property of a class.
+
+Fetch another coffee or tea and then let's add a static method to our `Book` class as well, so that we can create the new object statically instead.
```

Comment:
```suggestion
Grab another coffee or tea and then let's add a static method to our `Book` class as well, so that we can create the new object statically instead.
```